### ISSUE TITLE
Make Solder auto create the modslug directory

### ISSUE THREAD

**Issue by JamiePhonic** [TechnicPack/TechnicSolder#291]
I would like to see a feature implemented whereby when you add a new mod to solder, it will create the modslug directory for you if it doesnt already exist in your repository folder.

**Comment by TheDoctorsLife** [comment]
I see this sort of thing happening once mod uploads are there

**Comment by JamiePhonic** [comment]
i agree, but i still think having solder create the modslug directory for you would help people greatly, especially those with huge modpacks (200+ mods like some of mine are) who are just starting to use solder

**Comment by DeviPotato** [comment]
An issue with this is the fact that not everyone's modpack repositories are actually local to the Solder installation. I know for technical reasons I have to have the mods both stored locally (for quick hashing) and also mirrored on DropBox (where the launcher actually downloads the files from). Solder can't create directories in storage locations it has no control over.

**Comment by app1egenius93** [comment]
While what DeviPotato is true, it would still be possible to have it create the directories when possible. If not then oh well but otherwise, yes.

**Comment by Pyker** [comment]
As mentioned, this can only happen with local repos, and that's only if Solder has write access to it. Regardless, this requires a rewriting of how Solder interacts with repos before it can be a thing (probably using Laravel's storage drivers).
